File: 591/Main.java
import java.io.*;
import java.util.*;

public class Main {
   public static void main(String args[]) throws IOException {
      int testCase = 1;
      String line;
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

      while ((line = in.readLine()) != null) {
         if (line.charAt(0) == '0') break;
         int n = Integer.parseInt(line);
         int total = 0;
         int average = 0;
         int[] height = new int[n];
         StringTokenizer tokenizer = new StringTokenizer(in.readLine());

         for (int i = 0; i < n; i++) {
            height[i] = Integer.parseInt(tokenizer.nextToken());
            average += height[i];
         }

         average /= n;
         for (int i = 0; i < n; i++) {
            if (height[i] > average) {
               total += (height[i] - average);
            }
         }

         System.out.println("Set #" + testCase);
         System.out.println("The minimum number of moves is " + total + ".\n");
         ++testCase;
      }
   }
}

File: 591/main.c
#include <stdio.h>

int main(int argc, char* argv[])
{
   int height[50];
   int n, i, test_case, average, total;

   test_case = 1;
   while (scanf("%d", &n) == 1 && n > 0) {
      average = 0;
      for (i = 0; i < n; i++) {
         scanf("%d", &height[i]);
         average += height[i];
      }
      total = 0;
      average /= n;
      for (i = 0; i < n; i++) {
         if (height[i] > average) {
            total += (height[i] - average);
         }
      }
      printf("Set #%d\n", test_case);
      printf("The minimum number of moves is %d.\n\n", total);
      ++test_case;
   }

   return 0;
}
